### PR TITLE
Tests added

### PR DESCRIPTION
**Pull request by ShubhankarKG**
1. Models are tested extensively, hopefully remain stable until someone changes the model.
2. API is a WIP, but events (except for POST) and blogs are completely done.
3. The total tests stand at 52.

I hope we'll be able to setup Travis or CircleCI with these minimalistic tests.

### CODE REVIEW

**Comment by ShubhankarKG** [comment]
Fixes #13

**Comment by AliabbasMerchant** [comment]
Good job returning the proper status codes.  
I went through some of the tests. Look good.  

But there is a fundamental issue. You are directly modifying the DB while testing. This is NEVER EVER done. Use mock-ups.

So, use something like https://www.npmjs.com/package/mock-mongoose or https://github.com/nodkz/mongodb-memory-server, which are wrappers around mongoose/MongoDB

**Comment by AliabbasMerchant** [comment]
Please do the required changes, then I will review it.

And then I would like to add GitHub Actions for CI

**Comment by ShubhankarKG** [comment]
No prob. Probably add a second CI pipeline just in case. Github actions would work as well. I'll do the necessary work.

File: server/test/models/Blogs/delete_blogs_test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const assert = require('assert');
+const Blog = require("../../../src/models/Blog");
+
+let blog;
+
+
+
+describe("Delete blogs", () => {
+
```
Comment (AliabbasMerchant) [inline server/test/models/Blogs/delete_blogs_test.js:9]: Again, all this is not needed

File: server/test/models/Blogs/read_blogs_test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+let blog;
+
+
+
+describe("Reading blogs", () => {
+
+  beforeEach((done) => {
+    blog = new Blog({
+      blogTitle: "Test Blog Title",
```
Comment (AliabbasMerchant) [inline server/test/models/Blogs/read_blogs_test.js:12]: Again, unnecessary

File: server/test/models/Blogs/update_blogs_test.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+const assert = require('assert');
+const Blog = require("../../../src/models/Blog");
+
+let blog;
+let newBlog;
+
+describe("Updating blogs", () => {
+
```
Comment (AliabbasMerchant) [inline server/test/models/Blogs/update_blogs_test.js:8]: Again

File: server/test/models/Events/create_events_test.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const assert = require("assert");
+const Event = require("../../../src/models/Event");
```
Comment (AliabbasMerchant) [inline server/test/models/Events/create_events_test.js:2]: Again

File: server/test/models/Events/delete_events_test.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
+const assert = require('assert');
+const Event = require("../../../src/models/Event");
+
+let event;
+
```
Comment (AliabbasMerchant) [inline server/test/models/Events/delete_events_test.js:5]: I really appreciate your efforts, but again

File: server/test/models/Events/read_events_test.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+const assert = require('assert');
+const Event = require("../../../src/models/Event");
+
+let event;
```
Comment (AliabbasMerchant) [inline server/test/models/Events/read_events_test.js:4]: Again :sweat_smile:

File: server/test/models/Events/update_events.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
+const assert = require('assert');
+const Event = require("../../../src/models/Event");
+
+let event;
+let newEvent;
+
+describe("Updating Events", () => {
+
+  beforeEach((done) => {
+    event = new Event({
```
Comment (AliabbasMerchant) [inline server/test/models/Events/update_events.js:10]: :sweat_smile:

File: server/test/models/Glimpses/create_glimpses_test.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const assert = require('assert');
+const Glimpses = require("../../../src/models/GLimpses");
+
+describe('Testing Glimpses', () => {
```
Comment (AliabbasMerchant) [inline server/test/models/Glimpses/create_glimpses_test.js:4]: Same

File: server/test/models/Glimpses/delete_glimpses.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+const assert = require('assert');
+const Glimpses = require("../../../src/models/GLimpses");
+
+describe("Deleting glimpses", () => {
+  let glimpse;
+
+  beforeEach((done) => {
```
Comment (AliabbasMerchant) [inline server/test/models/Glimpses/delete_glimpses.js:7]: Same

File: server/test/models/Glimpses/read_glimpses_test.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+const assert = require('assert');
+const Glimpses = require("../../../src/models/GLimpses");
+
+let glimpse;
+
```
Comment (AliabbasMerchant) [inline server/test/models/Glimpses/read_glimpses_test.js:5]: Same

File: server/test/models/Glimpses/update_glimpses_test.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+const assert = require('assert');
+const Glimpses = require("../../../src/models/GLimpses");
+
+describe("Updating a glimpse", () => {
+  let glimpse;
+
+  beforeEach((done) => {
+    glimpse = new Glimpses({ eventName: "Test event" });
```
Comment (AliabbasMerchant) [inline server/test/models/Glimpses/update_glimpses_test.js:8]: Same

File: server/test/models/Users/create_user_test.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
+const assert = require("assert");
+const User = require("../../../src/models/User");
+
+describe('Testing Users', () => {
```
Comment (AliabbasMerchant) [inline server/test/models/Users/create_user_test.js:4]: Same

File: server/test/models/Users/delete_user_test.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
+const assert = require('assert');
+const User = require("../../../src/models/User");
+
+let user;
+
```
Comment (AliabbasMerchant) [inline server/test/models/Users/delete_user_test.js:5]: Same

File: server/test/models/Users/read_user_test.js [hunk 14]
```diff
@@ -0,0 +1,49 @@
+const assert = require('assert');
+const User = require("../../../src/models/User");
+
+let user;
+
+describe("Reading Users", () => {
```
Comment (AliabbasMerchant) [inline server/test/models/Users/read_user_test.js:6]: Same

File: server/test/models/Users/update_user_test.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
+const assert = require('assert');
+const User = require("../../../src/models/User");
+
+let user;
+let newUser;
+
+describe("Updating Users", () => {
+
```
Comment (AliabbasMerchant) [inline server/test/models/Users/update_user_test.js:8]: Same

File: server/test/api/blogs_test.js [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+
+  describe("/GET blogs", () => {
+    it("should return all blogs", (done) => {
+      chai
+        .request(app)
+        .get("/api/blogs")
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.a("array");
+          done();
```
Comment (AliabbasMerchant) [inline server/test/api/blogs_test.js:24]: Lets insert a few docs, and then test if we are getting them back or not

**Comment by AliabbasMerchant** [comment]
In effect, test for models are not needed. And you are sure, that testing is done in-memory?

File: server/test/api/users_test.js [hunk 17]
```diff
@@ -240,13 +254,15 @@ describe("Users", () => {
...
           .request(app)
           .post("/api/user")
           .send(user)
           .end((err, res) => {
+            expect(err).to.be.null;
             res.should.have.status(200);
-            res.body.should.be.a("object");
+            res.body.should.be.an("object");
             res.body.should.have.property("username").eql("Test user");
             res.body.should.have.property("password").eql(null);
```
Comment (AliabbasMerchant) [inline server/test/api/users_test.js:267]: A very important check :100:

File: server/src/controllers/Events.js [hunk 18]
```diff
@@ -1,78 +1,80 @@
...
+    const formURL = req.body.formURL;
+    const eventId = req.body.id;
+
+    try {
+      const event = await Event.findByIdAndUpdate(eventId, { form: formURL });
+      res.status(200).json({
+        message: "Form added successfully",
+      });
+    } catch (error) {
+      res.status(403).json({
```
[the remark this answers is not in the document]
Comment (ShubhankarKG) [reply server/src/controllers/Events.js:75]: The catch block indicates that the event was not found. in that case the form request was forbidden. Hence, the 403.

[the remark this answers is not in the document]
Comment (AliabbasMerchant) [reply server/src/controllers/Events.js:75]: ok

**Pull request merged**